apps/mobile-wallet/src/features/buy/BuyModal.tsx

```diff
@@ -0,0 +1,162 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { useEffect, useRef, useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { BackHandler, Platform } from 'react-native'
+import { Portal } from 'react-native-portalize'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import WebView, { WebViewNavigation } from 'react-native-webview'
+import styled, { useTheme } from 'styled-components/native'
+
+import AppText from '~/components/AppText'
+import Button from '~/components/buttons/Button'
+import BottomModal, { BottomModalProps } from '~/components/layout/BottomModal'
+import { ModalContent } from '~/components/layout/ModalContent'
+import ScreenTitle from '~/components/layout/ScreenTitle'
+import LinkToWeb from '~/components/text/LinkToWeb'
+import { useAppSelector } from '~/hooks/redux'
+import { InWalletTabsParamList } from '~/navigation/InWalletNavigation'
+import { selectAllDiscoveredAddresses } from '~/store/addressDiscoverySlice'
+import { selectDefaultAddress } from '~/store/addressesSlice'
+import { DEFAULT_MARGIN } from '~/style/globalStyle'
+import { showToast } from '~/utils/layout'
+
+interface BuyModalProps extends Omit<BottomModalProps, 'Content'> {}
+
+const BuyModal = (props: BuyModalProps) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<NavigationProp<InWalletTabsParamList>>()
+  const theme = useTheme()
+  const webViewRef = useRef<WebView>(null)
+  const insets = useSafeAreaInsets()
+  const defaultAddress = useAppSelector(selectDefaultAddress)
+  const allAddresses = useAppSelector(selectAllDiscoveredAddresses)
+  const [isDisclaimerAccepted, setIsDisclaimerAccepted] = useState(false)
+
+  const onAndroidBackPress = () => {
+    if (webViewRef.current) {
+      webViewRef.current.goBack()
+      return true // prevent default behavior (exit app)
+    }
+    return false
+  }
+
+  useEffect(() => {
+    if (Platform.OS === 'android') {
+      BackHandler.addEventListener('hardwareBackPress', onAndroidBackPress)
+      return () => {
+        BackHandler.removeEventListener('hardwareBackPress', onAndroidBackPress)
+      }
+    }
+  }, [])
+
+  const receivingAddress = defaultAddress?.hash || allAddresses[0]?.hash
+
+  if (!receivingAddress) {
+    showToast({
+      text1: t('Could not find a receiving address'),
+      text2: t('Cannot proceed because no address has been found to receive the funds.'),
+      type: 'error'
+    })
+    return null
+  }
+
+  const banxaURL =
+    'https://alephium.banxa-sandbox.com/' +
+    `?walletAddress=${receivingAddress}` +
```

Comment:
Shouldn't the user be allowed to configure the receiving address?

Reply:
I don't know. I'd say no, this would avoid mistakes and increase security.

Reply:
No, I mean, shouldn't the user select which of their addresses they want to use, like we do when sending/receiving funds?

Reply:
For our first implementation I'm tempted to say no.
- Let's reduce the pain and time needed to buy (there's the KYC etc.) by removing a step if we can
- Let's see what's the conversion rate
- Let's collect feedback

And then we'll reassess.